src/js/index.js

```diff
@@ -104,3 +108,38 @@ window.addEventListener('load', () => {
 
 	cascade.call(window, true);
 });
+
+const preference = localStorage.getItem('color-preference');
+if(preference=='dark')
+{
+	container[0].classList.add('dark');
+	mode.setAttribute("src","/assets/img/sun.png");
+	tooltip[0].textContent="Enable light mode";
+
+}
+else if(preference=='light')
+{
+	container[0].classList.remove('dark');
+		mode.setAttribute("src","/assets/img/moon.png");
+}
+
+mode.addEventListener("mouseover",function(){
+	tooltip[0].style.display='block';
+});
+
+mode.addEventListener("mouseout",function(){
+	tooltip[0].style.display='none';
+});
+
+
+ mode.addEventListener("click",function(){  
+	if(mode.getAttribute("src")=="/assets/img/moon.png")
+	{
+		localStorage.setItem('color-preference','dark');
+	}
+	else
+	{
+		localStorage.setItem('color-preference','light');
+	}
+	location.reload();
```

Comment:
Page reload is not needed!

Reply:
Clicking the color-switch changes the local storage value. If I don't reload the page then it will remain the same, since the decision about which theme to use is done on page load. :/